src/parse.jl

```diff
@@ -33,24 +33,25 @@ function parse_text(io, start_text::Int, end_text::Int)
     raw_btext = Array{UInt8}(undef, end_text - start_text + 1)
     read!(io, raw_btext)
     raw_text = String(raw_btext)
-    delimiter = raw_text[1]
+    # initialize iterator, save&skip the delimiter
+    delimiter, state = iterate(raw_text)
 
+    # container for the results
     text_mappings = Dict{String, String}()
-    # initialize iterator
-    iter_result = iterate(raw_text)
-    while iter_result !== nothing
-        i, state = iter_result
-
-        # found a new key, value pair
-        if i == '$'
-            # grab key and ignore escaped delimiters
-            key, state = grab_word(raw_text, state, delimiter)
-            # grab value and ignore escaped delimiters
-            value, state = grab_word(raw_text, state, delimiter)
-            # FCS keywords are case insensitive so force them uppercase
-            text_mappings["\$"*uppercase(key)] = value
+
+    while true
```

Comment:
Why not keep the original iteration format? AFAIK, I was using standard Julian iteration. Ref: https://julialang.org/blog/2018/07/iterators-in-julia-0.7/#iteration-in-julia-0-7

Reply:
Sorry for delay. The standard iterator pattern would be great IF it allowed cleanly peeking into the iterator and seeing whether it is on the end of input without consuming anything. I'll try to post a workaround in a few minutes.

Reply:
(Original reason for removal: removing the single first character kills several possible special cases later in the processing, especially if there's the "double separator" as the first character.)